src/message.py

```diff
@@ -67,6 +67,17 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return str(self)
 
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, Message):
+            return False
+        return (self.id == other.id and self.subject == other.subject 
+        and self.from_ == other.from_ and self.in_reply_to == other.in_reply_to 
+        and self.content == other.content and self.change_id == other.change_id
+        and self.archive_hash == other.archive_hash and self.children == other.children)
+
+    def __hash__(self) -> int:
+        return hash(self.id)
```

Comment:
I assume this was to make `Set[Message]` work.
I'm sorta OK with this, but at the same time, it feels a bit wrong that `hash(m1) == hash(m2)` can be true while `m1 != m2` (e.g. if we didn't update `.children`).

I had initially suggested we convert `Set[Message]` to `Set[str]` (of ids) so we could specifically avoid doing something like this.
I'd still sorta prefer that.

Reply:
Removed hash due to the reasons you mentioned. Changed Set[Message] to Dict[str, Message] in order to minimize database lookups without dealing with hashing of messages.